awx/main/tasks.py

```diff
@@ -2052,9 +2051,6 @@ def build_env(self, inventory_update, **kwargs):
         env['INVENTORY_SOURCE_ID'] = str(inventory_update.inventory_source_id)
         env['INVENTORY_UPDATE_ID'] = str(inventory_update.pk)
         env.update(STANDARD_INVENTORY_UPDATE_ENV)
-        plugin_name = inventory_update.get_inventory_plugin_name()
-        if plugin_name is not None:
-            env['ANSIBLE_INVENTORY_ENABLED'] = plugin_name
```

Comment:
I agree with the change to the `Job` runtime environment, but I don't actually agree with the change to the inventory update runtime environment.

You're on an entirely different level of hack to hit this there, to the point of severely undermining what the server is trying to do.